smithy-rules-engine/src/main/java/software/amazon/smithy/rulesengine/validators/AuthSchemesValidator.java

```diff
@@ -139,6 +140,34 @@ private Stream<ValidationError> validateSigv4(
                     propertyIs(authScheme, SIGNING_REGION, Literal::asString));
         }
 
+        private Stream<ValidationError> hasAllKeys(
+            Map<Identifier, Literal> map,
+            List<Identifier> keys,
+            SourceLocation sourceLocation
+        ) {
+            return keys.stream().flatMap(key -> {
+                if (!map.containsKey(key)) {
+                    return Stream.of(new ValidationError(ValidationErrorType.INVALID_AUTH_SCHEMES,
```

Comment:
Can you add a negative test that shows this behavior is working properly?

Reply:
added negative test. current negative tests do not invoke standalone validation, so added a separate folder for that